developer-experience/readme.md

```diff
@@ -0,0 +1,130 @@
+# Developer Experience
+
+Developer experience refers to the how easy or difficult it is for a developer to perform essential tasks needed to implement a change. A positive developer experience would mean these tasks are relatively easy for the team (see measures below).
+
+The essential tasks are identified below.
+
+Build - Verify that changes are free of syntax error and compile.
+Test - Verify that all automated tests pass.
+Start - Launch end-to-end to simulate execution in a deployed environment.
+Debug - Attach debugger to started solution, set breakpoints, step through code, and inspect variables.
+
+If effort is invested to make these activities as easy as possible, the returns on that effort will be multiply the longer the project runs and the larger the team.
+
+## Goals
+
+- Maximize the amount of time engineers spend on writing code that fulfills story acceptance and done-done criteria.
+- Minimize the amount of time spent manual setup and configuration of tooling
+- Minimize regressions and new defects by making end-to-end testing easy
+
+## Impact
+
+Developer experience can have a significant impact on the efficiency of the day to day execution of the team. A positive experience can pay dividends throughout the lifetime of the project; especially as new developers join the team.
+
+- Increased Velocity - Team spends less time on non-value-add activities such as dev/local environment setup, waiting on remote environments to test, and rework (fixing defects).
+- Improved Quality - When its easy to debug and test, developers will do more of it. This will translate to fewer defects being introduced.
+- Easier Onboarding & Adoption - When dev essential tasks are automated, there is less documentation to write and, subsequently, less to read to get started!
+
+## Measures
+
+### Time to First Debug (aka F5 Contract)
+
+Assuming a laptop/pc that has never run the solution, how long does it take to start debugging the solution end-to-end.
+
+### Time To First Commit
+
+How long does it take to make a change that can be verified/tested locally. A locally verified/tested change is one that passes test cases without introducing regression or breaking changes.
+
+## Participation
+
+The entire development team should take responsibility in ensuring positive

```

Comment:
Yes... but let's recommend an ongoing owner for this. Smells like a team lead responsibility. Could also be delegate formally to another senior team member.

Reply:
Thinking about this more @jplane, i put this rough outline together. What are your thoughts? The DevEx Champion represents your suggestion about formal delegation to an individual. cc @nmiodice 

### Tech Lead - Set the bar

The following are examples of how the Tech Lead might set the bar for dev experience

- Determines development environment (IDE, hosting, etc)
- Determines source control environment and number of repo's required
- Given development environment and repo structure sets expectations for team to meet in terms of steps to perform the essential dev tasks

### Senior Members - Assert Expectations

The following are examples of how the senior engineers on the team can assert the expectation set by the lead.

[`TODO: Add bullet list of examples`]

### Developer Experience Champion

[`TODO: Nominate an individual on the team to be the dev ex chamption`]

Reply:
Love it.

One additional suggestion... add another bullet under Tech Lead for "nominates/selects project DevEx champion... this could be the lead him/herself, or could be another senior team member"

Reply:
Also, per my comment below... let's outline a few basic responsibilities for the champion. I'm personally less interested in a cookie-cutter recipe... more interested in emphasizing iterative, proactive responsibility for care and feeding of devex for the duration of the project.

Reply:
Updated section. Please take a look and see if there are any other changes you think should be made. 

I agree of the emphasis. However, I would like to root this in a concrete practice to which the reader can relate. This can give some guidance on where to start then adapt as needed. Its not my intent to provide a recipe. More so to root in concrete examples. Hopefully this change reflects that. Let me know if you think it does not.